pytorch/training/docker/1.13/py3/Dockerfile.sagemaker.full.cpu

```diff
@@ -0,0 +1,294 @@
+RUN pip uninstall -y torch torchvision torchaudio torchdata
+
+# TODO: Install AWS-PyTorch and other torch packages here
+# PyTorch and other packages currently installed from pytorch-nightly pipeline
+RUN /opt/conda/bin/conda install \
+    pytorch=1.13.* \
+    torchvision \
+    torchaudio \
+    torchdata \
+    -c pytorch-nightly
+
+#TODO: Install PT S3 plugin
+RUN mkdir -p /etc/pki/tls/certs && cp /etc/ssl/certs/ca-certificates.crt /etc/pki/tls/certs/ca-bundle.crt
+
+RUN HOME_DIR=/root \
+ && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
+ && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
+ && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
+ && chmod +x /usr/local/bin/testOSSCompliance \
+ && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
+ && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
+ && rm -rf ${HOME_DIR}/oss_compliance* \
+ && rm -rf /tmp/tmp*
+
+# Starts framework
+CMD ["/bin/bash"]
+
+#################################################################
+#  ____                   __  __       _
+# / ___|  __ _  __ _  ___|  \/  | __ _| | _____ _ __
+# \___ \ / _` |/ _` |/ _ \ |\/| |/ _` | |/ / _ \ '__|
+#  ___) | (_| | (_| |  __/ |  | | (_| |   <  __/ |
+# |____/ \__,_|\__, |\___|_|  |_|\__,_|_|\_\___|_|
+#              |___/
+#  ___                              ____           _
+# |_ _|_ __ ___   __ _  __ _  ___  |  _ \ ___  ___(_)_ __   ___
+#  | || '_ ` _ \ / _` |/ _` |/ _ \ | |_) / _ \/ __| | '_ \ / _ \
+#  | || | | | | | (_| | (_| |  __/ |  _ <  __/ (__| | |_) |  __/
+# |___|_| |_| |_|\__,_|\__, |\___| |_| \_\___|\___|_| .__/ \___|
+#                      |___/                        |_|
+#################################################################
+
+FROM common AS sagemaker
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+ARG PYTHON
+
+#TODO: Redeclare build-time variables for PyTorch S3 plugin
+
+# TODO: Install SMDEBUG VERSION here
+
+ENV SAGEMAKER_TRAINING_MODULE=sagemaker_pytorch_container.training:main
+
+# TODO: Redeclare ARGs for PyTorch and other Torch packages here
+
+# uninstall if already present
+RUN pip uninstall -y torch torchvision torchaudio torchdata
+
+# TODO: Install AWS-PyTorch and other torch packages here
+
+RUN /opt/conda/bin/conda install \
+    pytorch=1.13.* \
+    torchvision \
+    torchaudio \
+    torchdata \
+    -c pytorch-nightly
+
+#TODO: Install PT S3 plugin
+
+RUN mkdir -p /etc/pki/tls/certs && cp /etc/ssl/certs/ca-certificates.crt /etc/pki/tls/certs/ca-bundle.crt
+
+WORKDIR /
+
+# Copy workaround script for incorrect hostname
+COPY changehostname.c /
+COPY start_with_right_hostname.sh /usr/local/bin/start_with_right_hostname.sh
+
+RUN chmod +x /usr/local/bin/start_with_right_hostname.sh
+
+# Install scikit-learn and pandas
+RUN conda install -y -c conda-forge \
+    scikit-learn \
+    pandas \
+    openssl==1.1.1o
```

Comment:
It can't be more general because there are multiple versions of openssl with different major versions. The current version in the image that is prone to vulnerabilities is 1.1.1q. Version 1.1.1o is the only version that addresses all 3 current CVEs that was raised. Attempts to do it at the same level as pyopenssl failed as the version 1.1.1q could not be overwritten with the latest version.

Reply:
Removed.